src/functions-and-arrays.js

```diff
@@ -1,41 +1,130 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(firstNumber, secondNumber) {
+
+  if (firstNumber > secondNumber) {
+    return firstNumber
+  } else {
+    return secondNumber
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(arrayOfWords) {
+
+  if (!arrayOfWords || arrayOfWords.length == 0) {
+    console.log("Please pass a non-empty array as function argument.")
+    return null;
+  }
+
+  let longestWord = arrayOfWords[0];
+  for (let i = 1; i < arrayOfWords.length; i++) {
+    if (arrayOfWords[i].length > longestWord.length) {
+      longestWord = arrayOfWords[i];
+    } else {
+      continue;
+    }
+  }
+  return longestWord;
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(arrayOfNumbers) {
+  let sum = 0;
+  if (arrayOfNumbers && arrayOfNumbers.length > 0) {
+    arrayOfNumbers.forEach(element => sum += element);
+  } else {
+    console.log("Please pass a non-empty array of integers as function argument.")
+  }
+  return sum;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(inputArray) {
+
+  if (!inputArray || inputArray.length == 0) {
+    return 0
+  }
+
+  let sum = 0;
+
+  for (i = 0; i < inputArray.length; i++) {
+    switch (typeof inputArray[i]) {
+      case 'number':
+        sum += inputArray[i];
+        break;
+      case 'string':
+        sum += inputArray[i].length;
+        break;
+      case 'boolean':
+        if (inputArray[i]) {
+          sum++;
+        }
+        break;
+      default:
+        throw new Error('Unsupported data type sir or ma\'am')
+        break;
+    }
+  }
+
+  return sum;
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(arrayOfNumbers) {
+
+  if (arrayOfNumbers.length > 0) {
+    let average = sumNumbers(arrayOfNumbers) / arrayOfNumbers.length;
+    return average;
```

Comment:
Here, to reduce your code even more, you can directly return the result, without declaring new variable :) 

`return sumNumbers(arrayOfNumbers) / arrayOfNumbers.length;`